File: tests/human/rust/dataset_sort_take_limit.rs
#[derive(Clone)]
struct Product { name: &'static str, price: i32 }

fn main() {
    let products = vec![
        Product { name: "Laptop", price: 1500 },
        Product { name: "Smartphone", price: 900 },
        Product { name: "Tablet", price: 600 },
        Product { name: "Monitor", price: 300 },
        Product { name: "Keyboard", price: 100 },
        Product { name: "Mouse", price: 50 },
        Product { name: "Headphones", price: 200 },
    ];

    let mut sorted = products.clone();
    sorted.sort_by(|a, b| b.price.cmp(&a.price));
    let selected: Vec<_> = sorted.into_iter().skip(1).take(3).collect();

    println!("--- Top products (excluding most expensive) ---");
    for item in selected {
        println!("{} costs $ {}", item.name, item.price);
    }
}

File: tests/transpiler/x/fs/dataset_sort_take_limit.fs
// Generated 2025-07-21 18:37 +0700

type Anon1 = {
    mutable name: string
    mutable price: int
}
type Anon2 = {
    mutable name: string
    mutable price: int
}
let products: Anon2 list = [{ name = "Laptop"; price = 1500 }; { name = "Smartphone"; price = 900 }; { name = "Tablet"; price = 600 }; { name = "Monitor"; price = 300 }; { name = "Keyboard"; price = 100 }; { name = "Mouse"; price = 50 }; { name = "Headphones"; price = 200 }]
let expensive = [ for p in List.take 3 (List.skip 1 (List.sortBy (fun p -> (-(p.price))) products)) do yield p ]
printfn "%s" (string "--- Top products (excluding most expensive) ---")
for item in expensive do
printfn "%s" (String.concat " " [string (item.name); string "costs $"; string (item.price)])
